### PR TITLE
route53 default to public hosted zone when both have the same domain

### PR DESCRIPTION
**Pull request by itomaldonado**
This should fix issue #61

### CODE REVIEW

File: src/main/kotlin/in/tazj/k8s/letsencrypt/acme/Route53Responder.kt [hunk 1]
```diff
@@ -72,6 +72,17 @@ class Route53Responder(val route53: AmazonRoute53) : DnsResponder {
                 .reduce { acc, zone ->
                     if (zone.name.length > acc.name.length) {
                         zone
+                    } else if (zone.name.length == acc.name.length) {
```
Comment (tazjin) [inline src/main/kotlin/in/tazj/k8s/letsencrypt/acme/Route53Responder.kt]: Should this compare the length or actually match the complete string? I haven't used private zones much, but I assume the actual name would be equal?

Comment (itomaldonado) [inline src/main/kotlin/in/tazj/k8s/letsencrypt/acme/Route53Responder.kt]: The actual name should be equal for this to work, so I guess that is a better comparison.

In AWS the only way to have two zones with the same name is if they are of different types (i.e. private vs. public).

I'll change it~

Comment (itomaldonado) [inline src/main/kotlin/in/tazj/k8s/letsencrypt/acme/Route53Responder.kt]: done.

**Comment by tazjin** [comment]
If you don't mind, I'll pull your changes and amend the commit messages to fit the project standard. I'll keep you as the author of course!

**Comment by tazjin** [comment]
In fact I just realised there probably isn't any situation where using a private zone here makes any sense, so we can just filter them out in the filter step above!

**Pull request closed without merging**